src/Components/WebView/WebView/src/StaticWebAssetsLoader.cs

```diff
@@ -0,0 +1,292 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Xml.Linq;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.Components.WebView
+{
+    internal class StaticWebAssetsLoader
+    {
+        internal const string StaticWebAssetsManifestName = "Microsoft.AspNetCore.StaticWebAssets.xml";
+
+        internal static IFileProvider UseStaticWebAssets(IFileProvider systemProvider)
+        {
+            using var manifest = GetManifestStream();
+            if(manifest != null)
+            {
+                return UseStaticWebAssetsCore(systemProvider, manifest);
+            }else
+            {
+                return systemProvider;
+            }
+
+            static Stream? GetManifestStream()
+            {
+                try
+                {
+                    var filePath = ResolveRelativeToAssembly();
+
+                    if (filePath != null && File.Exists(filePath))
+                    {
+                        return File.OpenRead(filePath);
+                    }
+                    else
+                    {
+                        // A missing manifest might simply mean that the feature is not enabled, so we simply
+                        // return early. Misconfigurations will be uncommon given that the entire process is automated
+                        // at build time.
+                        return null;
+                    }
+                }
+                catch
+                {
+                    return null;
+                }
+            }
+        }
+
+        internal static IFileProvider UseStaticWebAssetsCore(IFileProvider systemProvider, Stream manifest)
+        {
+            var webRootFileProvider = systemProvider;
+
+            var additionalFiles = StaticWebAssetsReader.Parse(manifest)
+                .Select(cr => new StaticWebAssetsFileProvider(cr.BasePath, cr.Path))
+                .OfType<IFileProvider>() // Upcast so we can insert on the resulting list.
+                .ToList();
+
+            if (additionalFiles.Count == 0)
+            {
+                return systemProvider;
+            }
+            else
+            {
+                additionalFiles.Insert(0, webRootFileProvider);
+                return new CompositeFileProvider(additionalFiles);
+            }
+        }
+
+        private static string? ResolveRelativeToAssembly()
+        {
+            var assembly = Assembly.GetEntryAssembly();
```

Comment:
Ideally this should be configurable, via config or something like that as the original version of it is. Otherwise it will fail in "obscure" scenarios like assembly renames and so on.